### PR TITLE
ADIOS2: Update to 2.9.1

### PR DESCRIPTION
**Pull request by eschnett**
Also disable building against HDF5

### CODE REVIEW

File: src/build.sh [hunk 1]
```diff
@@ -74,11 +86,11 @@ else
...
 
-if [ -n "${HAVE_CAPABILITY_HDF5}" ]; then
-    ADIOS2_HDF5_OPTS="-DHDF5_ROOT=${HDF5_DIR} -DADIOS2_USE_HDF5=ON"
-else
+# if [ -n "${HAVE_CAPABILITY_HDF5}" ]; then
+#     ADIOS2_HDF5_OPTS="-DHDF5_ROOT=${HDF5_DIR} -DADIOS2_USE_HDF5=ON"
+# else
     ADIOS2_HDF5_OPTS="-DADIOS2_USE_HDF5=OFF"
-fi
+# fi
```
Comment (rhaas80) [inline src/build.sh:93]: I'd remove the commented out code. If building against HDF5 is no longer supported then one can remove the dependency on HDF5 from `configuration.ccl`.

File: src/detect.sh [hunk 2]
```diff
@@ -10,10 +10,10 @@ if [ "$(echo ${VERBOSE} | tr '[:upper:]' '[:lower:]')" = 'yes' ]; then
 fi
 set -e                          # Abort on errors
 
-. $CCTK_HOME/lib/make/bash_utils.sh
+. "$CCTK_HOME/lib/make/bash_utils.sh"
```
Comment (rhaas80) [inline src/detect.sh:13]: make (actually `autoconf`'s macros) breaks if there is a space in the path to to Cactus so the double quotes are not going to really letting one have them (they don't hurt of course). At least not until we update `autoconf` to a version that supports spaces.
So is fine, but won't really do any good right now.

File: src/detect.sh [hunk 3]
```diff
@@ -42,13 +42,9 @@ fi
 ################################################################################
 
 ADIOS2_REQ_LIBS="adios2_cxx11 adios2_c adios2_core"
-if [ "$(echo ${ADIOS2_ENABLE_FORTRAN} | tr '[:upper:]' '[:lower:]')" = 'yes' ]; then
+if [ "${ADIOS2_ENABLE_FORTRAN}" = 'yes' ]; then
```
Comment (rhaas80) [inline src/detect.sh:45]: this removes recent required changes

File: src/detect.sh [hunk 4]
```diff
@@ -42,13 +42,9 @@ fi
...
     ADIOS2_REQ_LIBS="adios2_fortran $ADIOS2_REQ_LIBS"
 fi
-if [ "$(echo ${ADIOS2_ENABLE_SST} | tr '[:upper:]' '[:lower:]')" = 'yes' ] ; then
-    # core depends on adios2_evpath
-    ADIOS2_REQ_LIBS="$ADIOS2_REQ_LIBS adios2_evpath adios2_atl adios2_enet adios2_ffs adios2_dill"
-fi
```
Comment (rhaas80) [inline src/detect.sh:51]: these are required for BP5 output

File: src/detect.sh [hunk 5]
```diff
@@ -117,11 +113,14 @@ if [ -n "$ADIOS2_BUILD" -o -z "${ADIOS2_DIR}" ]; then
     # Fortran modules may be located in the lib directory
     ADIOS2_INC_DIRS="${ADIOS2_DIR}/include ${ADIOS2_DIR}/lib"
     ADIOS2_LIB_DIRS="${ADIOS2_DIR}/lib"
-    ADIOS2_LIBS="$ADIOS2_REQ_LIBS"
+    ADIOS2_LIBS="adios2_cxx11 adios2_c adios2_core"
+    if [ "$(echo ${ADIOS2_ENBABLE_FORTRAN} | tr '[:upper:]' '[:lower:]')" = 'yes' ]; then
+        ADIOS2_LIBS="adios2_fortran ${ADIOS2_LIBS}"
+    fi
```
Comment (rhaas80) [inline src/detect.sh:119]: this restores faulty code (note the typo in `adios_enBable_fortran`.

File: src/detect.sh [hunk 6]
```diff
@@ -10,10 +10,10 @@ if [ "$(echo ${VERBOSE} | tr '[:upper:]' '[:lower:]')" = 'yes' ]; then
 fi
 set -e                          # Abort on errors
 
-. $CCTK_HOME/lib/make/bash_utils.sh
+. "$CCTK_HOME/lib/make/bash_utils.sh"
 
 # Take care of requests to build the library in any case
-ADIOS2_DIR_INPUT=$ADIOS2_DIR
+ADIOS2_DIR_INPUT="$ADIOS2_DIR"
```
Comment (rhaas80) [inline src/detect.sh:16]: same as above, currently Cactus will always fail to build if there are spaces in its path (due to `autoconf`).

File: src/build.sh [hunk 7]
```diff
@@ -11,10 +11,22 @@ fi
...
+    
+# Define some environment variables
+export CC=${EXTERNAL_CC:-${CC}}
+export CXX=${EXTERNAL_CXX:-${CXX}}
+export F90=${EXTERNAL_F90:-${F90}}
+export LD=${EXTERNAL_LD:-${LD}}
+export CFLAGS=${EXTERNAL_CFLAGS:-${CFLAGS}}
+export CXXFLAGS=${EXTERNAL_CXXFLAGS:-${CXXFLAGS}}
+export F90FLAGS=${EXTERNAL_F90FLAGS:-${F90FLAGS}}
+export LDFLAGS=${EXTERNAL_LDFLAGS:-${LDFLAGS}}
```
Comment (rhaas80) [inline src/build.sh:23]: ideally this should be a separate commit (and in a perfect world we would not need it of course...).

**Review by rhaas80 — CHANGES_REQUESTED** [review]
I would split this into independent changes instead of a single commit that changes many things, some of which are not listed in the commit message

File: src/build.sh (hunk 1 above)
Comment (rhaas80) [reply src/build.sh:93]: have we actually tested if ADIOS2 with BP5 backend is indeed superior to ADIOS2 with HDF5 backend?

File: src/detect.sh (hunk 2 above)
Comment (rhaas80) [reply src/detect.sh:13]: see eg branch rhaas/spaces in cactus: https://bitbucket.org/cactuscode/cactus/commits/branch/rhaas%2Fspaces (which is not sufficient, at still fails to build due to some autoconf macro that constructs a full path to Cactus and cannot handle spaces)

**Comment by eschnett** [comment]
Thanks for the comments. I don't know where the unrelated changes come from. I'll re-do the pull request.

**Comment by rhaas80** [comment]
i pushed those changes to main very recently (without review, sorry). Without them I could not build BP5 support. Need to test the list of libraries on a cluster with pre-build dynamic libraries though, to see if they combined the long list of third party libraries into a single one or linked them to the adios core library.

**Comment by eschnett** [comment]
I am closing this PR. Will open a new, better one soon.
